server/src/main.rs

```diff
@@ -178,11 +185,34 @@ impl NodeStateService for NSServer {
     }
 }
 
+fn insert_sshfp(sshfpmap: &HashMap<String, KeySignature>, key_type: str, algorithm: u8, fingerprint: u8) {
+    sshfpmap.insert(key_type.to_string(), KeySignature {
+        algorithm_type: algorithm,
+        fingerprint_type: fingerprint,
+    })
+}
+//todo check hashlength to be proper
+fn add_to_sshfpmap(sshfpmap: &HashMap<String, KeySignature>, key_type: str) {
+    match key_type {
+        "rsa" => insert_sshfp(sshfpmap, "rsa", 1, 2),
+        "dsa" => insert_sshfp(sshfpmap, "dsa", 2, 2),
+        "ecdsa" => insert_sshfp(sshfpmap, "ecdsa", 3, 2),
+        "ed25519" => insert_sshfp(sshfpmap, "ed25519", 4, 2),
+        _ => println!("Invalid key_type provided: {}", key_type),
+    }
+}
+
 #[tokio::main]
 async fn main() -> Result<()> {
     env_logger::init();
     let opt = Opt::from_args();
 
+    let mut sshfpmap = HashMap::new();
+    add_to_sshfpmap(&sshfpmap, "rsa");
+    add_to_sshfpmap(&sshfpmap, "dsa");
+    add_to_sshfpmap(&sshfpmap, "ecdsa");
+    add_to_sshfpmap(&sshfpmap, "ed25519");
+
```

Comment:
This all feels incredibly roundabout. It doesn't look like there is any code to use it yet, but I'm sure we will talk about it more when you get further.